magnet_coils.py

```diff
@@ -276,6 +256,161 @@ def create_magnets(filaments, cross_section, meshing, logger):
     return vol_ids
 
 
+def clean_mag_data(filaments, tor_ext, r_avg, mag_len):
+    """Cleans filament data such that only filaments within the toroidal extent
+    of the model are included and filaments are sorted by toroidal angle.
+
+    Arguments:
+        filaments (list of list of list of float): list of filament coordinates.
+            Each filament is a list of coordinates.
+        tor_ext (float): toroidal extent to model (rad).
+        r_avg (float): average radial distance of magnets (cm).
+        mag_len (float): characteristic length of magnets.
+    
+    Returns:
+        sorted_fils (list of list of list of float): sorted list of filament
+            coordinates.
+    """
+    # Initialize data for filaments within toroidal extent of model
+    reduced_fils = []
+    # Initialize list of filament centers of mass for those within toroidal
+    # extent of model
+    com_list = []
+
+    # Define tolerance of toroidal extent to account for width of coils
+    # Multiply by factor of 2 to be conservative
+    tol = 2*np.arctan2(mag_len, r_avg)
+
+    # Compute lower and upper bounds of toroidal extent within tolerance
+    min_rad = 2*np.pi - tol
+    max_rad = tor_ext + tol
+
+    for fil in filaments:
+        # Compute filament center of mass
+        com = np.average(fil, axis = 0)
+        # Compute toroidal angle of each point in filament
+        phi_pts = np.arctan2(fil[:,1], fil[:,0])
+        # Compute bounds of toroidal extent of filament
+        min_phi = np.min(phi_pts)
+        max_phi = np.max(phi_pts)
+        # Ensure angles are positive
+        min_phi = (min_phi + 2*np.pi) % (2*np.pi)
+        max_phi = (max_phi + 2*np.pi) % (2*np.pi)
```

Comment:
Don't you have to do this before taking the min/max?  it seems you could have something that is less than 0 get identified as the minimum and then become the maximum once you've done this operation.

Reply:
Because `min_phi` and `max_phi` are being tested independently against `min_rad` and `max_rad`, it ends up not making a difference functionally but ensuring angles are positive before extracting the minimum and maximum seems like the cleaner way to do it.

Reply:
New commit incoming, just testing